src/components/encounter/encounter-form.component.tsx

```diff
@@ -86,11 +85,12 @@ const EncounterForm: React.FC<EncounterFormProps> = ({
   isSubmitting,
 }) => {
   const { t } = useTranslation();
+  const { newEncounterRole, isLoading: isLoadingEncounterRole } = useEncounterRole();
   const [fields, setFields] = useState<Array<FormField>>([]);
   const [encounterLocation, setEncounterLocation] = useState(null);
   const [encounterDate, setEncounterDate] = useState(formSessionDate);
   const [encounterProvider, setEncounterProvider] = useState(provider);
-  const [encounterRole, setEncounterRole] = useState(role);
+  const [encounterRole, setEncounterRole] = useState(newEncounterRole?.uuid);
```

Comment:
This won't correctly initialise the `encounterRole` state, it will introduce a regression for forms without a field capturing an encounter role.